webui/react/src/pages/ExperimentList.tsx

```diff
@@ -81,13 +81,119 @@ const ExperimentList: React.FC = () => {
     },
   );
   const initSorter = storage.getWithDefault(STORAGE_SORTER_KEY, { ...defaultSorter });
-  const [ pagination, setPagination ] = useState<Pagination>({ limit: initLimit, offset: 0 });
-  const [ total, setTotal ] = useState(0);
+  const [ experiments, setExperiments ] = useState<ExperimentItem[]>();
   const [ filters, setFilters ] = useState<ExperimentFilters>(initFilters);
-  const [ sorter, setSorter ] = useState(initSorter);
+  const [ isUrlParsed, setIsUrlParsed ] = useState<boolean>(false);
+  const [ pagination, setPagination ] = useState<Pagination>({ limit: initLimit, offset: 0 });
   const [ search, setSearch ] = useState('');
-  const [ experiments, setExperiments ] = useState<ExperimentItem[]>();
   const [ selectedRowKeys, setSelectedRowKeys ] = useState<string[]>([]);
+  const [ sorter, setSorter ] = useState(initSorter);
+  const [ total, setTotal ] = useState(0);
+
+  /*
```

Comment:
It might be a good idea to split those 2 effects out from the main component to increase code readability.

Something line `useParseAndUpdatePageURL(filters, pagination, search, sorter)`.

If you like the sound of that, what should be the file name? `webui/react/src/pages/ExperimentList.utils.ts`? Or `webui/react/src/pages/ExperimentList/utils.ts`?

Reply:
how about making it more generic so it can take in a config also to provide info on what fields and how to parse each field. Then this hook can just sit under `src/hooks`?

Reply:
We discussed via IM that is best to avoid "overengineering" this, we'll split the re-usable code when that will be needed (next page that will need a sharable URL).